datumaro/plugins/data_formats/datumaro_binary/mapper/media.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import struct
+from typing import Optional, Tuple
+
+from datumaro.components.errors import DatumaroError
+from datumaro.components.media import Image, MediaElement, MediaType, PointCloud
+
+from .common import Mapper, StringMapper
+
+
+class MediaMapper(Mapper):
+    @classmethod
+    def forward(cls, obj: Optional[MediaElement]) -> bytes:
```

Comment:
I am a bit confused that we are going to forward only the media.path through this. Correct?

Reply:
If so, how about forwarding media.path instead of separating media types?

Reply:
This acts a kind of factory for all kind of media types.
So, it has to write the media type first then go on to write its media structure (including path, and etc....) subsequently.

Reply:
I meant that we may be possible to locate forwarding `path` at here for all media types first, then the others (like size) can be forwarded according to each media types. What do you think?

Reply:
I added `backward_dict()` to `MediaElementMapper` and make `ImageMapper` and `PointCloudMapper` to inherit `MediaElementMapper` to use the common function `backward_dict()`. It will be a way to handle common member variable `path` across all media classes.